OverwatchHighlights/Extensions.cs

```diff
@@ -62,6 +62,16 @@ public static uint PeekUInt32(this BinaryReader br)
 			return value;
 		}
 
+		public static HighlightType ReadHighlightType64(this BinaryReader br)
+		{
+			const ulong HighlightTypeMask = 0x0830000000000000ul;
+			ulong value = br.ReadUInt64();
+			Debug.Assert((value & 0xFFFFFFFF00000000ul) == HighlightTypeMask);
```

Comment:
When the value is zero, the mask bits aren't set, so this check ought to be `(value & 0xFFFFFFFF00000000ul) == HighlightTypeMask || value == 0`